src/ngraph/runtime/cpu/mkldnn_emitter.hpp

```diff
@@ -1250,6 +1250,13 @@ namespace ngraph
                     Strides window_dilation_strides_adjusted;
 
                     mkldnn::algorithm convolution_algo = mkldnn_utils::get_conv_algo();
+
+                    if (node->get_input_element_type(0) != element::f32 &&
```

Comment:
This would also be true for element:f64, element::boolean, etc. Better to check for [u]int8 explicitly.

Reply:
right. We only want it to choose between auto and direct for f32 convolutions.

Reply:
@diyessi only fp32, u8 and s8 end up here. Rest go to reference kernels.